assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -388,6 +388,68 @@ const baseSelectors = {
 		return modules[ slug ];
 	} ),
 
+	/**
+	 * Gets module dependency names by slug.
+	 *
+	 * Returns a list of modules that depend on this module.
+	 * Returns `undefined` if state is still loading or if said module doesn't exist.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {(Array|undefined)} An array of dependency module names; `undefined` if state is still loading.
+	 */
+	getModuleDependencyNames: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const module = select( STORE_NAME ).getModule( slug );
+
+		// Return `undefined` if module with this slug isn't loaded yet.
+		if ( module === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found; return `[]` to signify the
+		// "not found" state.
+		if ( module === null ) {
+			return [];
+		}
+
+		// Module is found, return the names of the dependencies
+		const modules = select( STORE_NAME ).getModules();
+		return module.dependencies.map( ( dependencySlug ) => modules[ dependencySlug ].name );
```

Comment:
We need to filter out unavailable modules before mapping it to make sure we don't try to get a name from `undefined`. 

Plus, I think we need to use the module slug if the module name is empty: `modules[ dependencySlug ].name || dependencySlug`. It's not needed for the current modules, but since we are going to allow 3rd party developers to register their own modules, there might be a case when a module is registered with an empty name.

Reply:
I don't think module properties can be undefined per https://github.com/google/site-kit-wp/blob/fdc6a94a3951234c3b836f685d9ec65fcf37f8bf/assets/js/googlesitekit/modules/datastore/modules.js#L71

Reply:
@ivankruchkoff i mean a case when a 3rd part plugin registers its own module with dependencies like this: `[ 'analytics', 'foo', 'bar' ]`. Then, your code will find the Analytics module and will throw an error for `foo` because `modules['foo']` will result in `undefined`.

Reply:
Addressed.